### PR TITLE
Param validation 

### PR DESCRIPTION
**Pull request by deeghuge**
- Parameter Validation
- Changes to error detailed error for pvc creation failure

### CODE REVIEW

File: csiplugin/connectors/resources.go [hunk 1]
```diff
@@ -555,8 +555,10 @@ type CreateFilesetRequest struct {
 }
 
 type CreateMakeDirRequest struct {
-	UID string `json:"uid,omitempty"`
-	GID string `json:"gid,omitempty"`
+	UID   string `json:"uid,omitempty"`
+	GID   string `json:"gid,omitempty"`
+	USER  string `json:"user,omitempty"`
```
Comment (dunnevan) [inline csiplugin/connectors/resources.go]: what is the difference between UID and USER, and GID and GROUP?  I assume one is the number and one is the name?  Perhaps a comment here would be good.

**Review by dunnevan — DISMISSED** [review]
Added request for code comment.  Otherwise, LGTM

File: csiplugin/connectors/resources.go (hunk 1 above)
Comment (deeghuge) [reply csiplugin/connectors/resources.go]: > what is the difference between UID and USER, and GID and GROUP? I assume one is the number and one is the name? Perhaps a comment here would be good.

Agree. I have added the comments

**Review by dunnevan — APPROVED** [review]
Thanks LGTM

**Pull request merged**